packages/grid/x-data-grid/src/hooks/features/export/useGridPrintExport.tsx

```diff
@@ -262,15 +266,21 @@ export const useGridPrintExport = (
 
       await updateGridColumnsForPrint(options?.fields, options?.allColumns);
       apiRef.current.unstable_disableVirtualization();
+      await raf(); // wait for the state changes to take action
```

Comment:
This was already missing, but since we do no longer make a network request, the print displays a lot faster, which in the tests doesn't leave enough time for the API calls to render.